### PR TITLE
Replace gometalinter with golangci-lint

### PR DESCRIPTION
**Pull request by sapirpol**
### Checklist
- [X] Code compiles correctly
- [ ] Relevant tests were added (unit / contract / integration)
- [ ] Relevant logs were added
- [X] Formating and linting run locally successfully
- [X] All tests passing
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    - errcheck
+    - unused
+    - gosimple
+    - structcheck
+    - varcheck
+    - typecheck
+    - govet
+    - staticcheck
+    - ineffassign
+    - deadcode
```
Comment (tal-sapan) [inline .golangci.yml:17]: You already have this at the beginning

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+run:
+  deadline: 5m
+
+linters:
+  disable-all: true
+  enable:
```
Comment (tal-sapan) [inline .golangci.yml:6]: Is `gotypex` covered by `typecheck`?

Comment (tal-sapan) [inline .golangci.yml:6]: Is there an equivalent for `vetshadow`?

Comment (sapirpol) [reply .golangci.yml:6]: I'm not sure...
typecheck - Like the front-end of a Go compiler, parses and type-checks Go code.
gotype -x - Syntactic and semantic analysis in external test packages (similar to the Go compiler).

Comment (tal-sapan) [reply .golangci.yml:6]: External test packages are `*_test.go` files which are not in the same package as the original file (for example, if you have `fs` package and the test is in `fs_test` package).
Could you check if `typecheck` reports type errors in such files?

**Pull request merged**